spec/unit/win32/security_spec.rb

```diff
@@ -106,4 +106,29 @@
       expect { Chef::ReservedNames::Win32::Security.get_token_information_elevation_type(token) }.to raise_error(Chef::Exceptions::Win32APIError)
     end
   end
+
+  describe "self.lookup_account_name" do
+    let(:security_class) { Chef::ReservedNames::Win32::Security }
+
+    context "when FFI::LastError.error result is ERROR_INSUFFICIENT_BUFFER" do
+      it "does not raise the exception" do
+        expect(security_class).to receive(:error_insufficient_buffer?).and_return(true)
+        expect { security_class.lookup_account_name "system" }.to_not raise_error
+      end
+    end
+
+    context "when operation completed successfully and FFI::LastError.error result is NO_ERROR" do
+      it "does not raise the exception" do
+        expect(security_class).to receive(:error_insufficient_buffer?).and_return(true)
+        expect { security_class.lookup_account_name "system" }.to_not raise_error
+      end
+    end
```

Comment:
These specs also are hiding all the implementation details of `error_insufficient_buffer?` and should instead be stubbing `FFI::LastError.error` and testing against `NO_ERROR` and `ERROR_INSUFFICIENT_BUFFER` being set there.